core/modules/widgets/widget.js

```diff
@@ -557,6 +557,29 @@ Widget.prototype.allowActionPropagation = function() {
 	return true;
 };
 
+Widget.prototype.generateTransclusionFootprint = function() {
+	var parentTransclusionWidget = this.findParentTransclusionWidget(),
+	    node = this,
+	    footprint = node.parentWidget.children.indexOf(node);
+	while(node = node.parentWidget) {
+		if(node === parentTransclusionWidget) {
+			break;
+		} else if(node.parentWidget && node.parentWidget.children) {
+			footprint = footprint + "" + node.parentWidget.children.indexOf(node);
```

Comment:
I'm not sure if the `""` empty string is only there to "force" a string conversion, or if it was `"-"` which I think I'd prefer.

Reply:
Hi @pmario , it's there just to force string conversion, but thinking about what you mentioned above about when there are more than 10 siblings, I also want it to be "-" or another character.
I thought I'd use "-" to concatenate the state-qualifier with this string, that's why here a different character would be good, what do you think?

Reply:
yes, we can also just use "-" and create the id by `this.getStateQualifier() + "_" + this.generateTransclusionFootprint();` , then it will look like `728119345_0-12-4-0-0`

Reply:
I'm ok with this structure